subscription/src/main/java/org/killbill/billing/subscription/engine/core/DefaultSubscriptionBaseService.java

```diff
@@ -209,13 +213,20 @@ private boolean onPhaseEvent(final DefaultSubscriptionBase subscription, final S
                                               null;
             if (nextPhaseEvent != null) {
                 dao.createNextPhaseEvent(subscription, readyPhaseEvent, nextPhaseEvent, context);
-                return true;
+                phaseEventCreated = true;
             }
-        } catch (final SubscriptionBaseError e) {
+            
+            if(subscription.getCurrentPhase().getPhaseType() == PhaseType.FIXEDTERM) { 
```

Comment:
Shouldn't the logic be similar to what you had in `DefaultSubscriptionBaseApiService#getEventsOnCreation`, i.e:
1. If there is a next phase then create next phase event
2. Else if (this is the last phase) and this is a `FIXEDTERM` then insert the EXPIRED event

Reply:
Fixed